apps/postgres-new/components/app-provider.tsx

```diff
@@ -138,7 +138,17 @@ export default function AppProvider({ children }: AppProps) {
         if (isStartupMessage(message)) {
           const parameters = parseStartupMessage(message)
           if ('client_ip' in parameters) {
-            setConnectedClientIp(parameters.client_ip === '' ? null : parameters.client_ip)
+            // client disconnected
+            if (parameters.client_ip === '') {
+              setConnectedClientIp(null)
+              // we ensure we're not in a transaction block first
+              await db.sql`rollback;`.catch()
```

Comment:
`discard all` can't be run in a transaction block and it seems like after a `pg_dump` we still have a transaction opened somehow. 🤔

Reply:
How bizarre!